python/lsst/cp/pipe/defects.py

```diff
@@ -277,13 +291,18 @@ def runDataRef(self, dataRef, visitList):
 
         defectLists = {'dark': [], 'flat': []}
 
+        midTime = 0
+        filters = set()
+
         if self.config.mode == 'MASTER':
             if len(visitList) > 1:
                 raise RuntimeError(f"Must only specify one visit when using mode MASTER, got {visitList}")
-            dataRef.dataId['visit'] = visitList[0]
+            dataRef.dataId['expId'] = visitList[0]
```

Comment:
Requisite comment that processing should happen in a Task.run() method, and not the Task.runDataRef() method.

Reply:
Whilst appropriate, 1) this change is not relevant, and 2) it's hard for `cp_pipe` stuff operating on tons of visits, as you know. Will ignore this comment for now unless you shout 🙂